### PR TITLE
Add a default help script in Makefile

### PR DESCRIPTION
**Pull request by amritghimire**
A default target of help is added in make file which will list the
targets available in Makefile by parsing the Makefile itself. The source
of implementation for the auto documented makefile is added in the
comment itself.

### CODE REVIEW

**Comment by kabirbaidhya** [comment]
@amritghimire Can you add a screenshot of how the output of the commands look like now?

**Comment by amritghimire** [comment]
Sample Output on `make` command.
image

**Comment by kabirbaidhya** [comment]
@amritghimire 
1. Can you remove the last line `version commit@` - because that doesn't make sense here.  
2. Rename `target` to `command`.

File: Makefile [hunk 1]
```diff
@@ -2,34 +2,110 @@ SHELL :=/bin/bash
...
+			if (line_length <= 0) { \
+				line_length = ncol - indent - length(words[i]) - 1; \
+				printf "\n%*s ", -indent, " "; \
+			} \
+			printf "%s ", words[i]; \
+		} \
+		printf "\n"; \
+	}' \
+	| more $(shell test $(shell uname) = Darwin && echo '--no-init --raw-control-chars')
+	@echo -e $$(echo "version commit@" && git rev-parse --short HEAD)
```
Comment (kabirbaidhya) [inline Makefile]: Remove this.

File: Makefile [hunk 2]
```diff
@@ -2,34 +2,110 @@ SHELL :=/bin/bash
...
+# 	* append hold space (+ newline) to line
+# 	* replace newline plus comments by `---`
+# 	* print line
+# Separate expressions are necessary because labels cannot be delimited by
+# semicolon; see <http://stackoverflow.com/a/11799865/1968>
+.PHONY: help
+## Show this help message
+help:
+	@echo -e "Usage:\n\tmake <target>\n"
+	@echo "$$(tput bold)Available targets:$$(tput sgr0)"
```
Comment (kabirbaidhya) [inline Makefile]:
```suggestion
	@echo "$$(tput bold)Available commands:$$(tput sgr0)"
```

File: Makefile [hunk 3]
```diff
@@ -2,34 +2,110 @@ SHELL :=/bin/bash
...
+# 	* remove target prerequisites
+# 	* append hold space (+ newline) to line
+# 	* replace newline plus comments by `---`
+# 	* print line
+# Separate expressions are necessary because labels cannot be delimited by
+# semicolon; see <http://stackoverflow.com/a/11799865/1968>
+.PHONY: help
+## Show this help message
+help:
+	@echo -e "Usage:\n\tmake <target>\n"
```
Comment (kabirbaidhya) [inline Makefile]:
```suggestion
	@echo -e "Usage:\n\tmake <command>\n"
```

**Pull request merged**